### PR TITLE
feature(planner): support table statements in new planner

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- `DROP TABLE`
- `ALTER TABLE`
- `UNDROP TABLE`
- `RENAME TABLE`
- `TRUNCATE TABLE`
- `OPTIMIZE TABLE`
- `DESCRIBE TABLE`

## Changelog

- New Feature

## Related Issues

Closes https://github.com/datafuselabs/databend/issues/5774

### CODE REVIEW

File: query/src/sql/planner/binder/ddl/database.rs [hunk 1]
```diff
@@ -34,22 +35,26 @@ impl<'a> Binder {
...
-            .catalog
-            .as_ref()
-            .map(|catalog| catalog.name.clone())
-            .unwrap_or_else(|| self.ctx.get_current_catalog());
+        let AlterDatabaseStmt {
+            if_exists,
+            catalog,
+            database,
+            action,
+        } = stmt;
```
Comment (TCeason) [inline query/src/sql/planner/binder/ddl/database.rs:43]: let stmt= AlterDatabaseStmt {
            if_exists,
            catalog,
            database,
            action,
        };

**Review by TCeason — APPROVED** [review]

File: query/src/sql/planner/plans/scalar.rs [hunk 2]
```diff
@@ -31,6 +32,8 @@ pub trait ScalarExpr {
 
     fn used_columns(&self) -> ColumnSet;
 
+    fn is_deterministic(&self) -> bool;
```
Comment (leiysky) [inline query/src/sql/planner/plans/scalar.rs:35]: What is this for?

Comment (andylokandy) [reply query/src/sql/planner/plans/scalar.rs:35]: cluster key expression must be deterministic, so I've to check the property.

Comment (leiysky) [reply query/src/sql/planner/plans/scalar.rs:35]: @andylokandy Nice. By the way, you may need to add some check about deterministic cluster key for `CREATE TABLE` as well, which I haven't done before.

**Review by leiysky — APPROVED** [review]

File: common/meta/app/src/schema/table.rs [hunk 3]
```diff
@@ -174,8 +174,8 @@ pub struct TableMeta {
     pub engine: String,
     pub engine_options: BTreeMap<String, String>,
     pub options: BTreeMap<String, String>,
-    pub cluster_key: Option<String>,
-    // The vector of cluster keys.
+    pub current_cluster_key: Option<String>,
```
Comment (zhyass) [inline common/meta/app/src/schema/table.rs:177]: I think default_cluster_key is better than current_cluster_key. Because we have a default_cluster_key_id

**Pull request merged**